src/Kernel/IndexedSlot.class.st

```diff
@@ -48,12 +48,39 @@ IndexedSlot >> isVirtual [
 	^false
 ]
 
+{ #category : #'meta-object-protocol' }
+IndexedSlot >> object: anObject instVarAt: anIndex [
+	"Primitive. Answer a fixed variable in an object. The numbering of the 
+	 variables corresponds to the named instance variables. Fail if the index 
+	 is not an Integer or is not the index of a fixed variable. Essential for the
+	 debugger. See  Object documentation whatIsAPrimitive."
+
+	<primitive: 73>
+	"instead of copying the fail code here, just call the method that has it"
+	^thisContext object: anObject instVarAt: anIndex
+]
+
+{ #category : #'meta-object-protocol' }
+IndexedSlot >> object: anObject instVarAt: anIndex put: aValue [ 
+	"Primitive. Store a value into a fixed variable in the argument anObject.
+	 The numbering of the variables corresponds to the named instance
+	 variables.  Fail if the index is not an Integer or is not the index of a
+	 fixed variable.  Answer the value stored as the result. Using this
+	 message violates the  principle that each object has sovereign control
+	 over the storing of values into its instance variables. Essential for the
+	 debugger. See Object documentation whatIsAPrimitive."
+
+	<primitive: 74>
+	"instead of copying the fail code here, just call the method that has it"
+	^thisContext object: anObject instVarAt: anIndex put: aValue 
```

Comment:
Years ago I extracted all mirrors to MirrorPrimitives class. I think it can be used here. 
I wonder if there will be any performance difference comparing to such internal primitive call

Reply:
Yes, I was thinking: why not have a class Mirror that has the method on the class side, but that one can instantiate for an object, too (and provide the API on that instance, too).

This way we can use the class side methods here and not pay the cost to create a mirror instance.

Reply:
Having the mirror methods in a dedicated class is for sure the way to go: I want to add some methods so we can get the slots of an instance without sending messages. (I am experimenting with a "Reflective Object Graph Visitor" that uses the power of being able to annotate instance variables to denote sub-graphs...)